### PR TITLE
make p::d::Tria<1> similar to dim>1

### PR DESCRIPTION
**Pull request by davydden**

### CODE REVIEW

File: tests/distributed_grids/1d_grid.expect=build.output [hunk 1]
```diff
@@ -0,0 +1,106 @@
+
+DEAL:2d::hyper_cube
```
Comment (bangerth) [inline tests/distributed_grids/1d_grid.expect=build.output]: Is this actual output, or do you just need to provide *any* kind of file for the test to work? If the latter, just use an empty file.

Comment (davydden) [reply tests/distributed_grids/1d_grid.expect=build.output]: it looks like we need any file, including an empty one for `test.expected=build.output` to pass.

**Comment by davydden** [comment]
> Looks good with the exception of the one point.

fixed.

/run-tests

**Comment by davydden** [comment]
./run-tests

File: tests/distributed_grids/1d_grid.cc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+
+int main(int argc, char *argv[])
+{
+  Utilities::MPI::MPI_InitFinalize mpi_initialization (argc, argv, 1);
+
+  std::ofstream logfile("output");
+  deallog.attach(logfile);
+
+  parallel::distributed::Triangulation<dim> tr(MPI_COMM_WORLD);
```
Comment (masterleinad) [inline tests/distributed_grids/1d_grid.cc]: Where is `dim` defined? Shouldn't this just be:
```
parallel::distributed::Triangulation<1> tr(MPI_COMM_WORLD);
```

File: tests/distributed_grids/1d_grid.cc [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// Test that we can compile p::d::Tria<1>
+
+#include "../tests.h"
+#include <deal.II/distributed/tria.h>
+
+
+
+int main(int argc, char *argv[])
+{
+  Utilities::MPI::MPI_InitFinalize mpi_initialization (argc, argv, 1);
```
Comment (masterleinad) [inline tests/distributed_grids/1d_grid.cc]: If you add
```
   deal_II_exceptions::disable_abort_on_exception();
```
you can even make sure that we get an error message in `Debug` mode.

File: include/deal.II/distributed/tria.h [hunk 4]
```diff
@@ -959,16 +984,6 @@ namespace parallel
...
-      /**
-       * dummy settings
-       */
-      enum Settings
-      {
-        default_setting = 0x0,
-        mesh_reconstruction_after_repartitioning = 0x1,
-        construct_multigrid_hierarchy = 0x2
-      };
-
```
Comment (masterleinad) [inline include/deal.II/distributed/tria.h]: Any reason for moving this?

Comment (davydden) [reply include/deal.II/distributed/tria.h]: yes, it should be defined before constructor, right?

Comment (masterleinad) [reply include/deal.II/distributed/tria.h]: Yes, you're right.

File: tests/distributed_grids/1d_grid.expect=build.output [hunk 5]
```diff
@@ -0,0 +1,2 @@
+
+DEAL::Ok
```
Comment (masterleinad) [inline tests/distributed_grids/1d_grid.expect=build.output]: Does this test run now?

Comment (davydden) [reply tests/distributed_grids/1d_grid.expect=build.output]: it's not meant to run, it's only meant to build

Comment (davydden) [reply tests/distributed_grids/1d_grid.expect=build.output]: @tamiko the test output name is `1d_grid.expect=build.output`, but this test clearly runs:

```
distributed_grids/1d_grid.debug: ******    RUN failed    *******
```

Am I misunderstanding something in how files should be named?

Comment (masterleinad) [reply tests/distributed_grids/1d_grid.expect=build.output]: What I was intending to suggest is
```
 int main(int argc, char *argv[])
 {
   deal_II_exceptions::disable_abort_on_exception();
   Utilities::MPI::MPI_InitFinalize mpi_initialization (argc, argv, 1); 
   initlog();

   try
     {
       parallel::distributed::Triangulation<1> tr(MPI_COMM_WORLD);
     }
   catch (const std::exception &exc)
     {
       deallog << exc.what() << std::endl;
     }
 }
```
and run the test in `Debug` mode, i.e. rename the output file to `1d_grid.debug.output`.

**Pull request merged**